rust/core/src/error.rs

```diff
@@ -22,6 +22,10 @@ use std::{ffi::NulError, fmt::Display};
 
 use arrow_schema::ArrowError;
 
+use crate::constants;
+
+pub type AdbcStatusCode = u8;
```

Comment:
I guess we can't move this to the FFI crate because the TryFrom implementation has to be in this crate?

Reply:
As I wrote in the comments below, I'm wondering if this should be placed in the `constants` mod since it's defined in `adbc.h`.
https://github.com/apache/arrow-adbc/pull/3381#discussion_r2314257268

Reply:
Fine by me. I just don't want another layer of conversion/indirection.